src/Orchard.Web/Modules/Orchard.Autoroute/Services/AutorouteService.cs

```diff
@@ -44,7 +54,25 @@ public class AutorouteService : Component, IAutorouteService {
                 throw new ArgumentNullException("part");
             }
 
-            var pattern = GetDefaultPattern(part.ContentItem.ContentType).Pattern;
+            var itemCulture = _cultureManager.GetSiteCulture();
+
+            //if we are editing an existing content item
+            if (part.Record.Id != 0) {
+                ContentItem contentItem = _contentManager.Get(part.Record.ContentItemRecord.Id);
+                var aspect = contentItem.As<ILocalizableAspect>();
+
+                if (aspect != null) {
+                    itemCulture = aspect.Culture;
+                }
+            }
+
+            //if we are creating from a form post we use the form value for culture
+            HttpContextBase context = _httpContextAccessor.Current();
+            if (context.Request.Form["Localization.SelectedCulture"] != null) {

```

Comment:
This is a major code smell, accessing Request.Form from the AutorouteService. Is there no other way?